From 1aefd16964c30264c05d136cef3e1e7de5eb9c53 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Fri, 12 Sep 2014 07:46:43 +0100
Subject: [PATCH] sna: Use default monitor options on the first output

commit 6554cf0a69f04710b64f3488384fe5cb2748dc35
Author: Chris Wilson <chris@chris-wilson.co.uk>
Date:   Mon Aug 11 12:22:17 2014 +0100

    sna: Parse output options early during initialisation

rearranged the monitor query to before the num_outputs increment. The
result was that it choose the second output as the default and not the
intended first.

Bugzilla: https://bugs.gentoo.org/show_bug.cgi?id=522500
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
---
 src/sna/sna_display.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/sna/sna_display.c b/src/sna/sna_display.c
index f136c20..f6cfc80 100644
--- a/src/sna/sna_display.c
+++ b/src/sna/sna_display.c
@@ -3751,7 +3751,7 @@ sna_output_add(struct sna *sna, unsigned id, unsigned serial)
 	output->name = (char *)(output + 1);
 	memcpy(output->name, name, len + 1);
 
-	output->use_screen_monitor = config->num_output != 1;
+	output->use_screen_monitor = config->num_output != 0;
 	xf86OutputUseScreenMonitor(output, !output->use_screen_monitor);
 	assert(output->options);
 
-- 
2.1.0

